### PR TITLE
Maxdiff

### PR DESCRIPTION
**Pull request by Zhiwei5**
Maxdiff function & Test

### CODE REVIEW

File: Maxdiff.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+def max_difference(x):
+    diff_array = []
+    for index, number in enumerate(x):
+        # if index > 0:
```
Comment (anikamukherji) [inline Maxdiff.py]: I think you meant to delete this line

File: Maxdiff.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        if index > 0:
+            print(index)
+            diff = x[index] - x[index - 1]
+            diff_array.append(diff)
+            print(diff)
+            print(diff_array)
+
+    max_diff = max(diff_array)
```
Comment (anikamukherji) [inline Maxdiff.py]: So this works, but I think a better way of finding the max difference is to have a `curr_max` value that is initially 0, then is updated to the `max(curr_max, x[index] - x[index-1]`. It is just more efficient to update the max value along the way, rather than doing max on an entire array later

File: Maxdiff.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+def max_difference(x):
+    diff_array = []
+    for index, number in enumerate(x):
+        # if index > 0:
+        if index > 0:
+            print(index)
```
Comment (anikamukherji) [inline Maxdiff.py]: We need to remove all print statements before merging

File: Maxdiff.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+            diff = x[index] - x[index - 1]
+            diff_array.append(diff)
+            print(diff)
```
Comment (anikamukherji) [inline Maxdiff.py]: remove

File: Maxdiff.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+            print(diff_array)
```
Comment (anikamukherji) [inline Maxdiff.py]: remove

File: test_Maxdiff.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import pytest
+from Maxdiff import *
```
Comment (anikamukherji) [inline test_Maxdiff.py]: As a general rule about file naming, if "snake case" is used (where the words are separate with underlines), then all of the letters need to be undercase. Can you change the name of your module file to `test_max_difference.py` and `max_difference.py`? And then change the name of your functions to match?

File: test_Maxdiff.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def test_Maxdiff():
+    test_data1 = [1, 5, 6, 7, 12]
+    test_data2 = [3, 4, 5, 56, 45]
+    test_data3 = [6, 34, 23, 45, 57]
```
Comment (anikamukherji) [inline test_Maxdiff.py]: Good! I would add a test with some negative numbers, and make sure you test an empty array.

File: Maxdiff.py (hunk 2 above)
Comment (Zhiwei5) [reply Maxdiff.py]: That is a good idea. But what if the list we have is like[20, 17, 15, 13], if we set curr_max=0, then the function would return max diff to be 0. However, the true max diff should be -2.

Comment (anikamukherji) [reply Maxdiff.py]: Great point! So actually in that case, the max difference between adjacent numbers would be 3, since we should be finding the max of the absolute value of `x[index] - x[index-1]` (and 20-17 = 0). So we should really be doing `max(curr_max, math.abs(x[index] - x[index-1]))`. Feel free to split that onto more than one line if you think it makes it easier to read

Comment (anikamukherji) [reply Maxdiff.py]: Instead of using `math.fabs()`, you can insert an import statement in the function `from math import fabs` and then just use fabs(x[index] - x[index-1])

Comment (anikamukherji) [reply Maxdiff.py]: Also note that using `fabs` will change the return value to always being a `float` rather than sometimes an `int`. I actually think this is better, because we want the return value to always be constant, and using a float is more robust

**Comment by yz398** [comment]
Thank you so much! I will keep moving.
